File: portal/src/main/java/net/ownportal/portal/feed/Stream.java
package net.ownportal.portal.feed;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import net.ownportal.portal.source.Source;

@Entity
@Table(name = "streams")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Stream {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "source_id")
    private Source source;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id")
    private Group group;

    static Stream build(String name, Group group, Source source) {
        return Stream
            .builder()
            .name(name)
            .group(group)
            .source(source)
            .build();
    }
}

File: api/app/models/stream.rb
# == Schema Information
#
# Table name: streams
#
#  id         :integer          not null, primary key
#  name       :string
#  group_id   :integer
#  source_id  :integer
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  user_id    :integer
#
class Stream < ApplicationRecord
  belongs_to :group
  belongs_to :source
  belongs_to :user
end
